src/main/java/bio/terra/common/stairway/StairwayConfig.java

```diff
@@ -0,0 +1,59 @@
+package bio.terra.common.stairway;
+
+import static com.google.cloud.ServiceOptions.getDefaultProjectId;
+
+import bio.terra.common.kubernetes.KubeProperties;
+import bio.terra.common.kubernetes.KubeService;
+import bio.terra.stairway.Stairway;
+import bio.terra.stairway.exception.StairwayExecutionException;
+import java.util.Optional;
+import javax.inject.Provider;
+import org.apache.commons.lang.StringUtils;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+@EnableConfigurationProperties(value = {StairwayProperties.class, KubeProperties.class})
+public class StairwayConfig {
+
+  private final StairwayProperties stairwayProperties;
+  private final KubeProperties kubeProperties;
+
+  public StairwayConfig(StairwayProperties stairwayProperties, KubeProperties kubeProperties) {
+    this.stairwayProperties = stairwayProperties;
+    this.kubeProperties = kubeProperties;
+  }
+
+  @Bean
+  @Qualifier("stairway.cluster.name")
+  public String getStairwayClusterName(
+      KubeService kubeService, KubeProperties kubernetesProperties) {
+    return kubeService.getNamespace()
+        + Optional.ofNullable(kubernetesProperties.getNamespaceSuffix()).orElse(StringUtils.EMPTY)
+        + "--stairwaycluster";
+  }
+
+  @Bean
+  public Stairway getStairway(
+      ApplicationContext applicationContext,
+      @Qualifier("stairway.cluster.name") Provider<String> stairwayClusterNameProvider) {
```

Comment:
I think if we have to use Provider we are doing it wrong. I would rather move setting in Stairway from construction to initialization.

In fact, I would prefer a setup without a StairwayConfig and just with the lifecycle manager. Let it construct Stairway given the properties and initialize Stairway (having moved the settings to initialization).

The (eventually common) JobService can autowire StairwayLifecycleManager and use a getter for performing flight submits and status checks.